js/Ice/bidir/MockAlarmClock.ts

```diff
@@ -0,0 +1,30 @@
+
+import { Ice } from "@zeroc/ice";
+import { EarlyRiser } from "./AlarmClock.js";
+
+export class MockAlarmClock extends EarlyRiser.AlarmClock
+{
+    private _needMoreTime = true;
+    private _resolver: ((value: void | PromiseLike<void>) => void);
+
+    public StopPressed = new Promise<void>((resolve, reject) => this._resolver = resolve);
+
+
+    // Implements the abstract method ring from the AlarmClock class generated by the Slice compiler.
```

Comment:
I would add a comment like "can return a value or a future" since we don't have other demos showing off async servant methods.

Reply:
I updated the return type to be `EarlyRiser.ButtonPressed | Promise<EarlyRiser.ButtonPressed>`

Which matches what we generate in the TypeScript definition.

Reply:
I'd rather only return EarlyRiser.ButtonPressed and update the comment to be:
```
// Implements the abstract method ring from the AlarmClock class generated by the Slice compiler.
// Here, we provide a synchronous implementation that returns a value (EarlyRiser.ButtonPressed). We
// could also provide an async implementation that returns a promise (Promise<EarlyRiser.ButtonPressed> for
// this method).
```

Reply:
using `EarlyRiser.ButtonPressed | Promise<EarlyRiser.ButtonPressed>` is very typescript

Reply:
ok